src/rules/no-commonjs.js

```diff
@@ -27,10 +27,34 @@ function allowRequire(node, options) {
 // Rule Definition
 //------------------------------------------------------------------------------
 
+const schemaString = { enum: ['allow-primitive-modules'] }
+const schemaObject = {
+  type: 'object',
+  properties: {
+    allowPrimitiveModules: { 'type': 'boolean' },
+    allowRequire: { 'type': 'boolean' },
+  },
+  additionalProperties: false,
+}
 
 module.exports = {
   meta: {
     docs: {},
+
+    schema: {
+      anyOf: [
+        {
```

Comment:
I don't think so, options are always [an array of things](http://eslint.org/docs/developer-guide/working-with-rules#contextoptions).